### DISCUSSION TITLE
Enabling AWS IoT for compiling zbbridge firmware (refuses to compile the modules in)

### DISCUSSION THREAD

**Discussion by scotttag** [arendst/Tasmota#10566] [Q&A] [answered]
Hey folks,

This has been driving me up the wall for a few days now! I've been sucessfully flashing Tasmota to a ZBbridge and everything is working fine. I've been attempting to compile the firmware with AWS IOT enabled but it just won't compile it in...

I've followed the docs at https://tasmota.github.io/docs/AWS-IoT to add the definitions into user_config_override.h - I've even forced those definitions into my_user_config.h (uncommenting the defines in the AWS IoT section) but every time I compile (including clearing caches/build dir) - the firmware comes out at the exact same size and memory analysis in PlatformIO as the stock firmware (a 587K binary).

I've even tried forcing #define MQTT_TLS_ENABLED to true in my_user_config.h - but the compilation just seems to ignore everything and produce the exact same binary every time

I've flashed it anyway to test even though I knew it would be there, and TLSKey in the console simply returns a "command not found" error suggesting it is indeed missing the TLS libraries

Am I missing something basic on compiling AWS IoT and TLS MQTT into the zbbridge firmware?

**Comment by s-hadinger** [comment, marked as answer]
I'm afraid you did all this for nothing. The "tasmota-zbbridge" version already includes TLS and AWS IoT code in the pre-compiled binaries.

That said, customization of tasmota only works for "tasmota" version. If you compile "tasmota-zbbridge", it will mostly ignore your customizations.

**Reply by scotttag** [reply]
Haha, noooooo! Oh well, could be worse. I'm glad it's compiled in at least :) Thanks for the quick reply

I was just thrown off as following the guide (including using the provided CloudFormation template), it refuses to connect giving an error -4:
19:39:04 MQT: Attempting connection...
19:39:20 MQT: Connect failed to xxxx-ats.iot.eu-west-1.amazonaws.com:443, rc -4. Retry in 20 sec

Connectivity out to the host on 443 works fine on that network, verifiied from another device. I tried to turn the loggig up to max/more debug but nothing useful

**Reply by s-hadinger** [reply]
It looks like tls is not enabled. There is currently a bug in the web console that does not enable it when you check it. 

Use `SetOption103 1`

**Reply by scotttag** [reply]
Woo hoo, working perfectly now. Thank you so much! I never would've figured that out

Is it a known bug or should I pop an issue in?

**Reply by s-hadinger** [reply]
It's a bug I discovered this week. You don't need to do anything
